addons/edi_sale/models/edi_sale_request_document.py

```diff
@@ -95,13 +95,32 @@ def prepare(self, doc):
     @api.model
     def execute(self, doc):
         """Execute document"""
+        Partner = self.env['res.partner']
+
         super().execute(doc)
 
+        # Delete sale orders which no longer have any order lines.
+        SaleRequestRecord = self.sale_request_record_model(doc)
+        reqs = SaleRequestRecord.search([('doc_id', '=', doc.id)])
+        sales = reqs.mapped('sale_id')
+        to_remove = sales.filtered(lambda s: len(s.mapped('order_line')) == 0)
+        orderless_partners = to_remove.mapped('partner_id')
+        to_remove.unlink()
+        # Remove partners with no orders in the current document and no
+        # historic orders.
+        # TODO check this query is efficient.
+        domain = [('id', 'in', orderless_partners.mapped('id')),
+                  ('sale_order_ids', '=', False)]
+        partners_with_no_historic_orders = Partner.search(domain)
```

Comment:
Instead of a search, can we intersect `orderless_partners` with the partner records of the document?

Reply:
But wouldn't we end up trying to delete partners with empty orders in this import, but with orders from previous imports in the database?

Reply:
Yes, you are right.